addons/common/fnc_selectWeapon.sqf

```diff
@@ -23,30 +23,45 @@ Examples:
     (end)
 
 Author:
-    Sickboy, commy2
+    Sickboy, commy2, johnb43
 ---------------------------------------------------------------------------- */
 SCRIPT(selectWeapon);
 
-#define MAXIMUM_TRIES 100
-
+// Mode is not guaranteed to be config-case, so no case-sensitive comparisons possible
 params [["_unit", objNull, [objNull]], ["_weapon", "", [""]], ["_mode", "", [""]]];
 
 if (!local _unit) exitWith {false};
 
 private _vehicle = [vehicle _unit, _unit] select (_unit call CBA_fnc_canUseWeapon);
 
-private _index = 0;
+// Holster weapon
+if (_weapon == "") exitWith {
+    _unit action ["SwitchWeapon", _vehicle, _unit, 299];
+
+    currentMuzzle _unit == "" // return
+};
 
-if (_mode isEqualTo "") then {
-    while {_index < MAXIMUM_TRIES && {currentMuzzle _unit != _weapon}} do {
-        _unit action ["SwitchWeapon", _vehicle, _unit, _index];
-        _index = _index + 1;
+// If on foot or in FFV
+if (_unit isEqualTo _vehicle) then {
+    private _weaponState = (_unit weaponState _weapon) select [0, 3];
+
+    if (_mode != "") then {
+        _weaponState set [2, _mode];
     };
+
+    _unit selectWeapon _weaponState // return
 } else {
-    while {_index < MAXIMUM_TRIES && {currentMuzzle _unit != _weapon || {currentWeaponMode _unit != _mode}}} do {
-        _unit action ["SwitchWeapon", _vehicle, _unit, _index];
-        _index = _index + 1;
+    private _turretPath = _vehicle unitTurret _unit;
+    (weaponState [_vehicle, _turretPath, _weapon]) params ["", "_muzzle", "_currentMode"];
```

Comment:
```suggestion
    (weaponState [_vehicle, _turretPath, _weapon]) params ["_weapon", "_muzzle", "_currentMode"];
```
need this for selecting muzzles
e.g.
`[player, "AP"] call cba_fnc_selectWeapon` on badger IFV
want `_weapon` updated so it does
`selectWeaponTurret ["autocannon_40mm_CTWS",[0],"AP","player"]`